tests/tests.go

```diff
@@ -209,6 +209,22 @@ func initKubemacpoolParams(rangeStart, rangeEnd string) error {
 	return nil
 }
 
+func getVmFailCleanupWaitTime() time.Duration {
+	managerDeployment, err := testClient.KubeClient.AppsV1().Deployments(managerNamespace).Get(names.MANAGER_DEPLOYMENT, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred(), "Should succeed getting Deployment object")
+	Expect(len(managerDeployment.Spec.Template.Spec.Containers)).To(Equal(1))
```

Comment:
I think we should check if the manager container is there by its name instead of assuming there is one and only container in case we will add more containers in the future.

something like:
```
managerContainerName="manager"
...
Expect(len(managerDeployment.Spec.Template.Spec.Containers) >= 1).To(BeTrue())

managerContainer := corev1.Container{}
for container := managerDeployment.Spec.Template.Spec.Containers {
    if managerContainer.Name == managerContainerName {
       managerContainer = container
       break
    }
}
Expect(managerContainer).ToNot(BeEmpty())
Expect(managerContainer.Name).To(Equal(managerContainerName))
...
```
Also this code will be more reliable by not access an object by index|
(like we saw in other places in the code that this approach is prune to errors)

Reply:
totally agree